repo/fsrepo/fsrepo.go

```diff
@@ -359,14 +359,22 @@ func (r *FSRepo) Path() string {
 
 // SetAPIAddr writes the API Addr to the /api file.
 func (r *FSRepo) SetAPIAddr(addr ma.Multiaddr) error {
```

Comment:
If we fail to rename the temporary file, let's remove it.

Reply:
I would like to keep as it is. It feels like too much defensive programming. 

When users run the program next time, newly created tmp will override the existing tmp file. 

If you like to remove, I'll add code to remove the tmp file if rename fails

Reply:
Well, it is defensive programming. While this is unlikely to fail on Linux, renaming _can_ fail on Windows if some other user is accessing the API file at the same time. Yes, we'll end up deleting the temporary file later, but that isn't a reason to not at least _try_ to clean it up early.

Reply:
Made changes. Thanks for the explaining the reason.